sunpy/map/mapbase.py

```diff
@@ -611,14 +611,19 @@ def heliographic_longitude(self):
         heliographic_longitude = self.meta.get('hgln_obs', None)
 
         if heliographic_longitude is None:
-            if self._default_heliographic_longitude is None:
-                warnings.warn_explicit(
-                    "Missing metadata for heliographic longitude: "
-                    "assuming longitude of 0 degrees",
-                    Warning, __file__,
-                    inspect.currentframe().f_back.f_lineno)
-                self._default_heliographic_longitude = 0
-            heliographic_longitude = self._default_heliographic_longitude
+            if self.meta.get('crln_obs', None) is not None:
+                heliographic_longitude = float(self.meta['crln_obs']) - float(get_sun_L0(self.date))
```

Comment:
What was your thought process here? 

I would suggest also running code locally, so you can debug it.

Reply:
It was showing a type  error, so I thought since both values are returning values in degree type, hence it's not able to subtract, so I converted it to float.

Reply:
Unfortunately it doesn't seem to have worked. The tests are still failing. 

I would suggest picking a failing test and trying to run it with a `pdb` or `breakpoint` to find its cause.

Reply:
I would *guess* that you need to multiply `self.meta['crln_obs']` by `u.deg` to put it in degrees.

Reply:
> What was your thought process here?
> 
> I would suggest also running code locally, so you can debug it.

How can I do it locally?

Reply:
You could write a short script that imports and runs `Map` on the HMI  sample file or you could use `pytest` to run the tests.

Reply:
> What was your thought process here?
> 
> I would suggest also running code locally, so you can debug it.

How can I do it locally?

Reply:
> You could write a short script that imports and runs `Map` on the HMI sample file or you could use `pytest` to run the tests.

Okay, I will do that.

Reply:
> You could write a short script that imports and runs `Map` on the HMI sample file or you could use `pytest` to run the tests.

I was writing the script for testing but this error came up while importing `sunpy.map`

> ModuleNotFoundError: No module named 'numpy.core._multiarray_umath'

Reply:
@nabobalis can you help me with this?

Reply:
The numpy issue? I think you should check that all your installed packages are up to date.